### ISSUE TITLE
Allow users to configure if role has permission to use the admin UI client

### ISSUE THREAD

**Issue by lauriii** [jsdrupal/drupal-admin-ui#272]
Currently, by default, the admin UI loads for anonymous users but there's some errors because of missing permissions. Since we don't want to add those permissions necessarily to anonymous users, we probably want to restrict some users from using the app all together.

## Proposed resolution
Add new permission "access admin ui client" which is always checked before rendering the admin UI.

**Comment by dawehner** [comment]
> . Since we don't want to add those permissions necessarily to anonymous users, we probably want to restrict some users from using the app all together.

Shouldn't we be more defensive in the app code then?

**Comment by lauriii** [comment]
> Shouldn't we be more defensive in the app code then?

That is another approach we could take. However, I think this has to do with UX as well. I was wondering why the UI behaves in a strange way, some things missing on the UI and so on and it was because I wasn't logged in. I think this might be confusing for real users as well. Maybe this could be helped by showing the current user information somewhere?

**Comment by dawehner** [comment]
I'm wondering whether this is something we could ask the admin UI UX team for feedback. @ckrina, how would we indicate missing permissions for part of the application?

For me adding a permission would be nice temporarily though. As long we aren' the primary UI and think people should opt into it.

**Comment by ckrina** [comment]
I agree with Daniel with that having a permissions solution would be nice until we aren't the primary UI. 
For small interactions with content it's not always needed a complex UI and the frontend solution might be able to handle it without forcing a change of context.

About how to indicate missing permissions, I'd say to show a warning message saying you might face problems because of missing permissions is what makes more sense. How important or sticky this message should be I think it might depend on what you really can't do. Another way might be to clearly indicate if you're logged in or not.

Anyway, I think a final solution should take into account what we assume all users will need to have (enabled JS?) to use our admin, and maybe it's a broader discussion.

**Comment by dawehner** [comment]
Maybe as first step we could document at least which permissions are good to have https://github.com/jsdrupal/drupal-admin-ui/issues/298
